sdk/iot/core/README.md

```diff
@@ -55,6 +55,43 @@ int main()
 }
 ```
 
+### Properties
+
+Included in the IoT SDK are helper functions to form and manage properties for IoT Hub services. Implementation starts by using the `az_iot_hub_client_properties_init()` API. The user is free to intitialize using an empty, but appropriately sized, span to later append properties or an already populated span containing a properly formated property buffer. "Properly formatted" properties follow the form `{key}={value}&{key}={value}`.
+
+Below is an example use case of appending properties.
+
+```C
+//FOR SIMPLICITY THIS DOES NOT HAVE ERROR CHECKING. IN PRODUCTION ENSURE PROPER ERROR CHECKING.
+void my_property_func()
+{
+  //Allocate a span to put the properties
+  uint8_t property_buffer[64];
+  az_span property_span = az_span_init(property_buffer, 0, sizeof(property_buffer));
+  
+  //Initialize the property struct with the span
+  az_iot_hub_client_properties props;
+  az_iot_hub_client_properties_init(&props, property_span);
+  //Append properties
+  az_iot_hub_client_properties_append(&props, AZ_SPAN_FROM_STR("key"), AZ_SPAN_FROM_STR("value"));
+  //At this point, you are able to pass the `props` to other API's with property parameters.
+}
+```
+
+Below is an example use case of initializing an already populated property span.

```

Comment:
Shall we mention that the properties must be RFC 2396-encoded as per https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-mqtt-support#sending-device-to-cloud-messages?

Reply:
I can mention it in the doxygen header docs.

Reply:
Good point btw

Reply:
[Updated](https://github.com/Azure/azure-sdk-for-c/pull/483/commits/020e7b3b9f2840c55fa6378cba14897c0b0983ae)